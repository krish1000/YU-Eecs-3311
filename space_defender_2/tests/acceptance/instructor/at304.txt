--Test enemy fighter various projectile damage
toggle_debug_mode

play(5,19,1,2,2,2,2)
setup_select(3)
setup_next(1)
setup_select(4)
setup_next(1)
setup_select(3)
setup_next(2)
setup_next(1)


move(A,1)
pass
move(A,2)
move(A,3)
move(A,4)
