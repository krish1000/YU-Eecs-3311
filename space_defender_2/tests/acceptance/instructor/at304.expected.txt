  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,19,1,2,2,2,2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(3)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Snipe
->setup_next(1)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:None
->setup_select(4)
  state:armour setup, debug, ok
  1:None
    Health:50, Energy:0, Regen:1/0, Armour:0, Vision:0, Move:1, Move Cost:0
  2:Light
    Health:75, Energy:0, Regen:2/0, Armour:3, Vision:0, Move:0, Move Cost:1
  3:Medium
    Health:100, Energy:0, Regen:3/0, Armour:5, Vision:0, Move:0, Move Cost:3
  4:Heavy
    Health:200, Energy:0, Regen:4/0, Armour:10, Vision:0, Move:-1, Move Cost:5
  Armour Selected:Heavy
->setup_next(1)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Standard
->setup_select(3)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Armoured
->setup_next(2)
  state:setup summary, debug, ok
  Weapon Selected:Snipe
  Armour Selected:Heavy
  Engine Selected:Armoured
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:250/250, energy:200/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[C,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:250/250, energy:180/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,19]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [A,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Fighter(id:1) spawns at location [A,19].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:250/250, energy:196/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[A,1]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,13]
  Projectile:
    [-1,<]->damage:100, move:10, location:[A,12]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [A,1], doubling regen rate.
  Enemy Action:
    A Fighter(id:1) moves: [A,19] -> [A,13]
      A enemy projectile(id:-1) spawns at location [A,12].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A S  _  _  _  _  _  _  _  _  _  _  <  F  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(A,2)
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:163/250, energy:190/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[A,2]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,10]
  Projectile:
    [-2,<]->damage:20, move:3, location:[A,9]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,12] -> [A,2]
  Starfighter Action:
    The Starfighter(id:0) moves: [A,1] -> [A,2]
      The Starfighter collides with enemy projectile(id:-1) at location [A,2], taking 87 damage.
  Enemy Action:
    A Fighter(id:1) moves: [A,13] -> [A,10]
      A enemy projectile(id:-2) spawns at location [A,9].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A _  S  _  _  _  _  _  _  <  F  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(A,3)
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:167/250, energy:188/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[A,3]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,9]
  Projectile:
    [-2,<]->damage:20, move:3, location:[A,6]
    [-3,<]->damage:50, move:6, location:[A,8]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-2) moves: [A,9] -> [A,6]
  Starfighter Action:
    The Starfighter(id:0) moves: [A,2] -> [A,3]
  Enemy Action:
    A Fighter(id:1) moves: [A,10] -> [A,9]
      A enemy projectile(id:-3) spawns at location [A,8].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A _  _  S  _  _  <  _  <  F  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(A,4)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:127/250, energy:186/200, Regen:4/8, Armour:13, Vision:16, Move:6, Move Cost:10, location:[A,4]
      Projectile Pattern:Snipe, Projectile Damage:1000, Projectile Cost:20 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,8]
  Projectile:
    [-4,<]->damage:50, move:6, location:[A,7]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-2) moves: [A,6] -> [A,3]
      The projectile collides with Starfighter(id:0) at location [A,3], dealing 7 damage.
    A enemy projectile(id:-3) moves: [A,8] -> [A,3]
      The projectile collides with Starfighter(id:0) at location [A,3], dealing 37 damage.
  Starfighter Action:
    The Starfighter(id:0) moves: [A,3] -> [A,4]
  Enemy Action:
    A Fighter(id:1) moves: [A,9] -> [A,8]
      A enemy projectile(id:-4) spawns at location [A,7].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19
    A _  _  _  S  _  _  <  F  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
